src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/EditAndContinue/EditAndContinueProvider.cs

```diff
@@ -7,78 +7,217 @@
+    internal class EditAndContinueProvider : IVsENCRebuildableProjectCfg, IVsENCRebuildableProjectCfg2, IVsENCRebuildableProjectCfg4, IDisposable
     {
-        private readonly ILanguageServiceHost _host;
+        private ILanguageServiceHost _host;
 
         [ImportingConstructor]
         public EditAndContinueProvider(ILanguageServiceHost host)
         {
             _host = host;
         }
 
-        // AbstractProject implements IVsENCRebuildableProjectCfg2 and IVsENCRebuildableProjectCfg4 only
-        [ExportProjectNodeComService(typeof(EncInterop.IVsENCRebuildableProjectCfg2), typeof(EncInterop.IVsENCRebuildableProjectCfg4))]
-        [AppliesTo(ProjectCapability.EditAndContinue)]
-        internal object EditAndContinueService
-        {
-            get
-            {
-                return _host.HostSpecificEditAndContinueService;
-            }
-        }
-
-        // Managed ENC always uses IVsENCRebuildableProjectCfg2.
-        // Although we don't implement the methods of IVsENCRebuildableProjectCfg, it is important that we implement the interface and return VSConstants.E_NOTIMPL.
-        // This is how the EncManager recognizes the project system that supports EnC.
+        // NOTE: Managed ENC always calls through IVsENCRebuildableProjectCfg2/IVsENCRebuildableProjectCfg4.
+        // We implement IVsENCRebuildableProjectCfg as this used to sniff the project for EnC support.
         public int ENCRebuild(object in_pProgram, out object out_ppSnapshot)
         {
             out_ppSnapshot = null;
-            return VSConstants.E_NOTIMPL;
+            return HResult.NotImplemented;
         }
 
         public int BelongToProject(string in_szFileName, ENC_REASON in_ENCReason, int in_fOnContinue)
         {
-            return VSConstants.E_NOTIMPL;
+            return HResult.NotImplemented;
         }
 
         public int ENCComplete(int in_fENCSuccess)
         {
-            return VSConstants.E_NOTIMPL;
+            return HResult.NotImplemented;
         }
 
         public int CancelENC()
         {
-            return VSConstants.E_NOTIMPL;
+            return HResult.NotImplemented;
         }
 
         public int ENCRelink([In, MarshalAs(UnmanagedType.IUnknown)] object pENCRelinkInfo)
         {
-            return VSConstants.E_NOTIMPL;
+            return HResult.NotImplemented;
         }
 
         public int StartDebugging()
         {
-            return VSConstants.E_NOTIMPL;
+            return HResult.NotImplemented;
         }
 
         public int StopDebugging()
         {
-            return VSConstants.E_NOTIMPL;
+            return HResult.NotImplemented;
         }
 
         public int SetENCProjectBuildOption([In] ref Guid in_guidOption, [In, MarshalAs(UnmanagedType.LPWStr)] string in_szOptionValue)
         {
-            return VSConstants.E_NOTIMPL;
+            return HResult.NotImplemented;
+        }
+
+        public int StartDebuggingPE()
```

Comment:
I have a slight preference for using explicit implementation for cases like this, but it's not a big deal either way.

Reply:
👍